client/state/data-layer/wpcom/me/settings/index.js

```diff
@@ -83,14 +83,19 @@ export function saveUserSettingsFailure( { settingsOverride }, error ) {
 		];
 	}
 
-	return updateUserSettingsFailure( settingsOverride, error );
+	return [
+		errorNotice( error.message || translate( 'There was a problem saving your changes.' ), {
+			id: 'save-user-settings-failure',
```

Comment:
Using a different ID for failure and success notices carries an unwanted ambiguous notice behavior that's a regression when compared to production. To repro:

* In this branch, go to `/me`
* Make changes to the form
* Go offline and save. You'll see an error notice.
* Go online and save. You'll see a success notice, but the error notice will still remain.
* In production, the error notice is removed when a success one is added.
* A mirror version of the same issue will happen if an error occurs on a save after a successful save.

The easiest way to fix this is to keep the same notice ID for both success and error notices, for example `save-user-settings` for both success and error notices.

Reply:
Good call, I'll make sure to fix these 👍🏻

Reply:
Fixed via f73dbc0